src/core/Switch/Switch.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { Typography } from '@material-ui/core';
+import { CheckCircle, Cancel } from '@material-ui/icons';
+import { ButtonFilled } from '../Button/ButtonFilled';
+import { SwitchBaseProps } from './base';
+import { useStyles } from './styles';
+
+type Variant = 'default' | 'error' | 'success' | undefined;
+
+interface SwitchProps extends SwitchBaseProps {
+  variant?: Variant;
+}
+
+const Switch: React.FC<SwitchProps> = ({ variant }) => {
```

Comment:
Can you change the names Switch to ButtonGroup...Switch would be another component